platform/engineering/design-docs/2020-12-14-cms-content-caching-strategy.md

```diff
@@ -0,0 +1,186 @@
+
+If the current branch is `master`, also get the version that the `master` alias points to. If the master alias doesn't match the version of the checksum alias:
+- [Update the `master` alias](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/lambda/update-alias.html) to the version currently aliased with the checksum.
+- Synchronously invoke the new version of the function (aliased with `master`).
+
+Pull the cache from S3 that corresponds to the checksum (or `master` if on the `master` branch) for the build step.
+
+### Code Location
+
+Initially, the Lambda function and its supporting code will live in the `vets-website`.
+Once the content build is separated into the `content-build` repo, they will be moved there.
+
+Handler source code, bundling, and upload (`vets-website`):
+- Lambda function: `src/platform/lambdas/content-cache.js`
+- Webpack: `config/webpack.content-cache.js`
+- GitHub Actions: `.github/workflows/content-build.yml`
+- Function upload script: `scripts/upload-content-cache.sh`
+
+The Lambda function will be provisioned in the `devops` repo.
+- Terraform: `terraform/environments/global/lambda.tf`
+
+### Testing Plan
+
+The GitHub Actions pipeline will include a job for building and updating the function. This will alert us to any issues with building the function and provisioning it in AWS.
+
+The prototype for the function will simply download the data from the Drupal server. This is to verify that it can connect to the CMS at all.
+Once that is confirmed, the function will be further developed to cache the downloaded data to S3.
+
+When the function is fully written, we will monitor the log output in AWS CloudWatch to observe whether it works correctly.
+
+### Logging
+
+GitHub Actions will output logs for the build and versioning of the function.
+Log output from the function (from calling `console` methods) will be stored in CloudWatch.
+
+### Debugging
+
+One can start diagnosing any issues using the logs mentioned above.
+
+The Webpack build of the function can be run locally to debug issues with the build.
+
+To locally validate that the code properly pulls the Drupal data, the function can be run in a script with some modifications to disable the parts that are specific to the Lambda environment.
+
+Once the data is cached, it can be downloaded and used to build content. That build can be compared with a content build that pulls fresh data from Drupal with a SOCKS connection.
+
+### Caveats
+
+To be determined.
+
+### Security Concerns
+
+The function is not expected to take much bandwidth on the Drupal server, so there should be no concern about denial of service. It will be scheduled to make a request only every 5 or 10 minutes.
+
+Without any user input, it should not be susceptible to malicious requests either.
+
+### Privacy Concerns
+
+There are no privacy concerns as no user data is involved.
+
+### Open Questions and Risks
```

Comment:
Can we list the team we'll need to reach out to (or steps we need to take) to answer these questions incase someone else needs to pick up this doc? If we can do the research now, let's try and do that or at least a plan for when we need these answers by in order to move forward